src/Tools/NodeDocumentationMarkdownGenerator/AssemblyHandler.cs

```diff
@@ -0,0 +1,387 @@
+                var nodeModelType = dynamoCoreAss.GetType("Dynamo.Graph.Nodes.NodeModel");
+
+                var functionGroups = new Dictionary<string, Dictionary<string, FunctionGroup>>(new LibraryPathComparer());
+                
+                foreach (var path in assemblyPaths)
+                {
+                    try
+                    {
+                        var functionDescriptors = new List<FunctionDescriptor>();
+
+                        if (new FileInfo(path).Extension == ".ds")
+                        {
+                            var dsDescriptors = GetFunctionDescriptorsFromDs(pathManager, path);
+                            if (dsDescriptors != null)
+                            {
+                                functionDescriptors.AddRange(dsDescriptors);
+                            }
+                        }
+
+                        else
+                        {
+                            Assembly asm = mlc.LoadFromAssemblyPath(path);
+                            AssemblyName name = asm.GetName();
+
+                            if (NodeModelAssemblyLoader.ContainsNodeModelSubTypeReflectionLoaded(asm, nodeModelType))
+                            {
+                                FileInfosFromNodeModels(asm, nodeModelType, ref nodeSearchModel);
+                                continue;
+                            }
+
+                            var dllDescriptors = GetFunctionDescriptorsFromDll(pathManager, asm);
+                            if (dllDescriptors != null)
+                            {
+                                functionDescriptors.AddRange(dllDescriptors);
+                            }
+                        }                  
+
+                        foreach (var descriptor in functionDescriptors)
+                        {
+                            Dictionary<string, FunctionGroup> fptrs;
+                            if (!functionGroups.TryGetValue(descriptor.Assembly, out fptrs))
+                            {
+                                fptrs = new Dictionary<string, FunctionGroup>();
+                                functionGroups[descriptor.Assembly] = fptrs;
+                            }
+
+                            string qualifiedName = descriptor.QualifiedName;
+                            FunctionGroup functionGroup;
+                            if (!fptrs.TryGetValue(qualifiedName, out functionGroup))
+                            {
+                                functionGroup = new FunctionGroup(qualifiedName);
+                                fptrs[qualifiedName] = functionGroup;
+                            }
+                            functionGroup.AddFunctionDescriptor(descriptor);
+                        }
+                    }
+                    catch (Exception)
+                    {
+                        continue;
+                    }
+                }
+
+                if (functionGroups.Count > 0)
```

Comment:
huh? this seems odd - won't this be true only once? 
What case is this for?

Reply:
hm, not sure what you mean by only once, yes it will either be true or false once as its only checked once.

The reason im checking is so I don't try to add function descriptors (ZT nodes) to the NodeSearchModel if there arent any.
This can happen in scenarios where you are only scanning `NodeMoldel` libraries.